### ISSUE TITLE
tree_cmd option is ignored?

### ISSUE THREAD

**Issue by vbauerster** [ul/kak-tree#11]
None of `tree-*` commands respect my config, which has some blacklist for go file types.
I've `tree_cmd` option set in my kakrc and it expands correctly: 
```
set-option global tree_cmd "kak-tree -c %val{config}/kak-tree.toml"
```
This is content of my `kak-tree.toml`:
```
[filetype.go]
blacklist = ["nil", "block", "identifier"]
```
For example, if I issue `tree-node-sexp`, I still can see blacklisted items.

**Comment by ul** [comment]
`tree-node-sexp` ignores white/blacklist by design (we could parametrize it to support both modes though). Do you have problems with blacklist when using other commands?

**Comment by vbauerster** [comment]
Yes I do, for example `:tree-select-children nil<ret>` selects nil occurrences, despite it's blacklisted.
I would much appreciate if `tree-node-sexp` could drop blacklisted items, because it's too noisy by default. My workflow is first to run `tree-node-sexp` on selection in order to acquire definition node needed, then run `:tree-select-* var_node<ret>`.

**Comment by vbauerster** [comment]
IMHO it's not necessary for `tree-select-*` commands family to obey filter rules, but `tree-node-sexp` must obey in first place.

**Comment by vbauerster** [comment]
Also I wish `tree-node-sexp` had `-lock` switch, to freeze its menu.

I use `kak-lsp` as well with `gopls` which spawns to many unnecessary errors, so most of the times menu of `tree-node-sexp` gets overridden by `gopls`.

**Comment by ul** [comment]
Yes, that sounds reasonable. Also, I think we should improve the way blacklisted nodes are handled in general, it might worth not only scanning for visible nodes but removing invisible nodes from the tree in the first place (and lifting up their visible descendants). It will make navigation result more intuitive.

On an unrelated note, you can reduce noise from kak-lsp by overriding `lsp-show-error` command to filter our harmless errors from `gopls`.
